apis/python/src/tiledbsoma/soma_dense_nd_array.py

```diff
@@ -153,148 +146,72 @@ def is_sparse(self) -> Literal[False]:
+            return pa.Tensor.from_numpy(
+                arrow_tbl.column("data").to_numpy().reshape(target_shape)
             )
-        )
 
-    def read_as_pandas_all(
+    def read_numpy(
         self,
+        coords: SOMADenseNdCoordinates,
         *,
-        row_ids: Optional[Sequence[int]] = None,
-        col_ids: Optional[Sequence[int]] = None,
-        set_index: Optional[bool] = False,
-    ) -> pa.RecordBatch:
+        result_order: Optional[SOMAResultOrder] = None,
+    ) -> np.ndarray:
         """
-        This is a convenience method around ``read_as_pandas``. It iterates the return value from ``read_as_pandas`` and returns a concatenation of all the record batches found. Its nominal use is to simply unit-test cases.
+        Read a user-specified dense slice of the array and return as an Numpy ``ndarray``.
         """
-        dataframes = []
-        generator = self.read_as_pandas(
-            row_ids=row_ids,
-            col_ids=col_ids,
-            set_index=set_index,
+        return cast(
+            np.ndarray, self.read_tensor(coords, result_order=result_order).to_numpy()
         )
-        for dataframe in generator:
-            dataframes.append(dataframe)
-        return pd.concat(dataframes)
 
-    def write(
+    def write_tensor(
         self,
-        # TODO: rework callsites with regard to the very latest spec rev
-        # coords: Union[tuple, tuple[slice], NTuple, List[int]],
-        coords: Any,
+        coords: SOMADenseNdCoordinates,
         values: pa.Tensor,
     ) -> None:
         """
-        Write an Arrow.Tensor to the persistent object. As duplicate index values are not allowed, index values already present in the object are overwritten and new index values are added.
+        Write subarray, defined by ``coords`` and ``values``. Will overwrite existing
+        values in the array.
 
-        :param coords: location at which to write the tensor
+        Parameters
+        ----------
+        coords - per-dimension tuple of scalar or slice
+            Define the bounds of the subarray to be written.
 
-        :param values: an Arrow.Tensor containing values to be written. The type of elements in ``values`` must match the type of the SOMADenseNdArray.
+        values - pyarrow.Tensor
+            Define the values to be written to the subarray.  Must have same shape
+            as defind by ``coords``, and the type must match the SOMADenseNdArray.
         """
-
         with self._tiledb_open("w") as A:
             A[coords] = values.to_numpy()
 
+    def write_numpy(self, coords: SOMADenseNdCoordinates, values: np.ndarray) -> None:
+        """ "
+        Write a numpy ``ndarray`` to the user specified coordinates
+        """
+        self.write_tensor(coords, pa.Tensor.from_numpy(values))
+
     # ----------------------------------------------------------------
+    #
+    # TODO: this code seems obsolete given the current API. Can we port the `io` package
```

Comment:
would you like to file a tracking issue in GH or shall i?

Reply:
I'll create one (see the additional items above that are "similar")

Reply:
#361